mewarpx/mewarpx/utils_store/init_restart_util.py

```diff
@@ -3,36 +3,94 @@
 """
 import os
 import shutil
-import glob
 import logging
 
 logger = logging.getLogger(__name__)
 
+DEFAULT_CHECKPOINT_NAME = "checkpoint"
 
-default_checkpoint_name = "checkpoint"
 
-def clean_old_checkpoints(prefix=default_checkpoint_name, directory="diags",
-                          num_to_keep=0):
+def get_sorted_checkpoints(checkpoint_directory, checkpoint_prefix):
+    """Function to return valid checkpoints for restarting or removing
+    checkpoints.
+
+    Also warns if any checkpoints with .old. exist, which shouldn't in normal
+    workflow.
```

Comment:
At present we don't clean up an existing `diags` folder when starting a run as we do in `metools`. I think that would probably be a good addition to this PR - whenever a fresh run (not a restart) is started, we should delete `diags` if present. What do you think?

Reply:
Cleanest way to do it we think:
- Determine `self.restart` where it is in the code.
- Between that and initialization, `if not self.restart` then remove all `diags*` directories
- At initialization, all diagnostics would create folders with an afterinit python hook. diags folders created before init would be incorrect programming.
Let's do this in a separate PR, I'll make a monday.com card.